components/support/test/src/main/java/mozilla/components/support/test/rule/Helpers.kt

```diff
@@ -0,0 +1,57 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+@file:Suppress("INVISIBLE_MEMBER", "INVISIBLE_REFERENCE") // for TestMainDispatcher
+
+package mozilla.components.support.test.rule
+
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.TestCoroutineScheduler
+import kotlinx.coroutines.test.TestDispatcher
+import kotlinx.coroutines.test.TestResult
+import kotlinx.coroutines.test.TestScope
+import kotlinx.coroutines.test.internal.TestMainDispatcher
+import kotlinx.coroutines.test.runTest
+import kotlin.reflect.full.companionObject
+import kotlin.reflect.full.companionObjectInstance
+import kotlin.reflect.full.memberProperties
+
+/**
+ * `coroutines.test` default timeout to use when waiting for asynchronous completions of the coroutines
+ * managed by a [TestCoroutineScheduler].
+ */
+private const val DEFAULT_DISPATCH_TIMEOUT_MS = 60_000L
+
+/**
+ * Convenience method of executing [testBody] in a new coroutine running with the
+ * [TestDispatcher] previously set through [Dispatchers.setMain].
+ *
+ * Running a test with a shared [TestDispatcher] allows
+ * - newly created coroutines inside it will automatically be reparented to the test coroutine context.
+ * - leveraging an already set strategy for entering launch / async blocks.
+ * - easier scheduling control.
+ *
+ * @see runTest
+ * @see Dispatchers.setMain
+ * @see TestDispatcher
+ */
+@OptIn(ExperimentalCoroutinesApi::class)
+fun runTestOnMain(
```

Comment:
This is a drive-by, but it wasn't easy to find the definition of this in the GitHub review since it forces you to load all the diffs because of the 195 file changes. Couldn't we have name this `runBlocking` to save ourselves a lot of refactoring?

Reply:
My goal as per https://github.com/Kotlin/kotlinx.coroutines/blob/master/kotlinx-coroutines-test/MIGRATION.md was to refactor out `runBlocking` and `runBlockingTest`.
Seeing again these mentioned in tests would seem like a big code smell for me. 
And in that I think this would actually slow down development and review - just because we see something that should't be there. But maybe has reason to be there.

`runTestOnMain` is clearer in it's intent. It is intended to only be used when `Dispatchers.setMain` was previously called, otherwise it will throw an exception and fail the test.
It is just a quicker way of writing `mainCoroutineRule.scope.runTest` or `runTest(mainCoroutineRule.testDispatcher.scheduler)`.
In the majority of cases the main dispatcher is not set and `runTest` would work just fine.
Then there are the exceptional cases where `runBlocking` needs to be used - when `need to interact with dispatchers that execute on other threads, like Dispatchers.IO or Dispatchers.Default`.

Reply:
I see how the diff would've been much easier, but also agree that `runBlockingTest` is an oversimplified name and confusing. So, OK with `runTestOnMain` to indicate that the main dispatcher is used, and therefore needs to be set in the test via `mainCorountineRule`.